Sources/Fuzzilli/Compiler/Compiler.swift

```diff
@@ -987,11 +987,53 @@ public class JavaScriptCompiler {
 
             if unaryExpression.operator == "typeof" {
                 return emit(TypeOf(), withInputs: [argument]).output
+            } else if unaryExpression.operator == "delete" {
+                let argumentExpression = unaryExpression.argument.expression!
+
+                if case .memberExpression(let memberExpression) = argumentExpression {
+                    let obj = try compileExpression(memberExpression.object)
+                    let isGuarded = memberExpression.isOptional
```

Comment:
I think this could use a comment. I guess you are effectively turning
```
delete a?.b;
```
into
```
try { delete a.b; } catch {}
```
?
I think that makes sense, but a short comment might be nice :)

Reply:
Here is an example of before and after:

### Before
```
const nestedObj = { a: { b: 2 } };
delete nestedObj.a?.c;
```

### After
```
const o1 = {
    "b": 2,
};
const o2 = {
    "a": o1,
};
o2.a?.c;
const t6 = o2.a;
delete t6?.c;
```

So, no try catch enveloping which is nice in my opinion :)

Reply:
Looks good! Though were does the `o2.a?.c;` line come from? Is that maybe a bug in our lifting, or is that property access also there in the IL?

Reply:
Ah yes, good catch! This happened because 

` 
let argument = try compileExpression(unaryExpression.argument)
` 
was called, even though `argument` was not used in the delete case. This is fixed now. I will be looking more carefully at the FuzzIL translation in the future!

Reply:
Ah great thanks, and no worries, this is what code reviews are for :)